syft/frameworks/torch/crypto/spdz.py

```diff
@@ -14,14 +14,15 @@ def spdz_mul(
             self. Equivalent to calling self.child.
         other_shares: a dictionary <location_id -> PointerTensor) of shares corresponding
             to the tensor being multiplied by self.
-        cypto_provider: an AbstractWorker which is used to generate triples
+        crypto_provider: an AbstractWorker which is used to generate triples
         field: an interger denoting the size of the field
         """
     locations = list(shares.keys())
     shares_shape = shares[locations[0]].shape
     other_shape = other_shares[locations[0]].shape
     triple = crypto_provider.generate_triple(cmd, field, shares_shape, other_shape, locations)
     a, b, c = triple
+    a, b, c = a.child, b.child, c.child
```

Comment:
Why are the children added back?

Reply:
The way I'm reasoning is that all objects a, b or c should be AdditiveSharedTensor or MultiPointerTensoir, (all without wrappers)
I'm pretty confident this is a good way to proceed, I just made sure I had the good object in hands, but this line should not be there and crypto_provider.generate_triple shoudl return the correct type

Reply:
So the setup earlier was that everything in this method was actually a dictionary of pointer tensors. Has that changed?

Reply:
Yes I think it was not the best choice -- well it was good to have it run quickly; but it forces to always iterate manually while we have protocols to handle classic operations transparently

Reply:
I think we can re-rewrite the spdz protocol making the hypothesis we only handle AdditiveSharedTensor and MultiPointerTensor. That's what I did for snn, but I added functionalities so that they would play together nicely.

Reply:
I've just fixed the snn thing (so relu, eq, comp) works, I'm planning on opening another PR with code cleaned and correctly committed

Reply:
Sounds good to me. I have an open pr with decompose inplemented. Just need to write tests for it

Reply:
Maybe you can compare with the version of decompose implemented here also, it looks like it's also working :)